crates/nu-parser/src/parse_keywords.rs

```diff
@@ -3484,7 +3497,14 @@ pub fn parse_const(working_set: &mut StateWorkingSet, spans: &[Span]) -> (Pipeli
                     let rhs_type = rvalue.ty.clone();
 
                     if let Some(explicit_type) = &explicit_type {
-                        if !type_compatible(explicit_type, &rhs_type) {
+                        // see `parse_let`
+                        let compatible = if let Type::Any = rhs_type {
+                            true
+                        } else {
+                            type_compatible(explicit_type, &rhs_type)
+                        };
```

Comment:
this probably needs to be removed unless a `const` can do runtime coercion which would make it not const :).